src/Campaigns/Routes/Traits/RestResponses.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Give\Campaigns\Routes\Traits;
+
+use Give\Framework\Exceptions\Primitives\Exception;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+trait RestResponses
+{
+    /**
+     * @unreleased
+     */
+    public function notFoundResponse($message, $response = 'rest_not_found'): WP_REST_Response
+    {
+        return new WP_REST_Response(
+            [
+                'status' => 404,
+                'response' => $response,
+                'body_response' => [
+                    'message' => html_entity_decode(
+                        $message
+                    ),
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function badRequestResponse(Exception $e, $response = 'rest_bad_request'): WP_REST_Response
+    {
+        return new WP_REST_Response(
+            [
+                'status' => 400,
+                'response' => $response,
+                'body_response' => [
+                    'message' => html_entity_decode(
+                        sprintf(
+                            esc_html__('%s. Search the logs at Donations > Tools > Logs for a more specific cause of the problem.',
```

Comment:
Should this messaging be moved to the client (instead of being part of the route)?

Reply:
Resolved in 7f64906a2d650e4af1d6f95a6ba444ff62404ec4 and ea7a3d193d985b101ce808c5311ba7e3cd16e0ee